### PR TITLE
fltk: made used abi version of the build package configurable, set default to 10308.

### PR DESCRIPTION
**Pull request by bjoernrennfanz**
- Made used ABI_VERSION configurable with abi_version setting.
- Set abi_version to 10308, this fixes issues, when trying to compile samples from fltk github repository.

Specify library name and version:  **fltk/1.3.8**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/fltk/all/conanfile.py
Comment (SpaceIm) [inline recipes/fltk/all/conanfile.py:28]: I strongly advise to not hardcode this value. abi version should match fltk version by default, here you are hardcoding the abi version of a specific fltk version, it won't scale.

Comment (bjoernrennfanz) [reply recipes/fltk/all/conanfile.py]: I have changed that according to your advise. Thank you!

**Review by sblkr — APPROVED** [review]

**Review by jonaskarp — APPROVED** [review]

**Comment by bjoernrennfanz** [comment]
@SpaceIm Can you take a look at the latest changes? Thank you!

**Comment by prince-chrismc** [comment]
> * this fixes issues, when trying to compile samples from fltk github repository.

Can you provide more details as to what the error was and why this is required please, seems unusual to need this if the project is version well

***

I appreciate the enthusiasm approving but if you don't plan of contributing regularly please refrain from spamming please 🙏

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
Thanks for the link and the excellent summary 😀 👍 

You might want to consider updating this recipe in the near future for v2 support which is currently failing 😅

**Comment by prince-chrismc** [comment]
Don't merge master, it dismiss reviews for the bots custom logic

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**
